lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -21,11 +47,17 @@ def initialize(serializer, options = {})
         def serializable_hash(options = nil)
           options ||= {}
 
-          if serializer.respond_to?(:each)
-            serializable_hash_for_collection(options)
-          else
-            serializable_hash_for_single_resource(options)
-          end
+          hash =
+            case
```

Comment:
I don't really care what the style guide says, honestly, if we agree on something that makes sense to us.  It does mention 'leveraging' the return value of if/else and case statements.

Is this a blocker?